src/pybind/mgr/dashboard/frontend/src/app/shared/services/refresh-interval.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RefreshIntervalService {
+  // Observable sources
+  private intervalDataSource = new BehaviorSubject(null);
+
+  // Observable streams
+  intervalData$ = this.intervalDataSource.asObservable();
+
+  private intervalTime: number;
+  intervalID: number;
+
+  constructor() {
+    const initialInterval = parseInt(sessionStorage.getItem('dashboard_interval'), 10) || 5000;
+    this.setRefreshInterval(initialInterval);
+  }
+
+  setRefreshInterval(interval: number) {
+    this.intervalTime = interval;
+    sessionStorage.setItem('dashboard_interval', interval.toString());
+
+    window.clearInterval(this.intervalID);
```

Comment:
What about using RxJS approach (instead of the old 'window.setInterval') ?
What are your thoughts on this @tspmelo ?
```
--- a/src/pybind/mgr/dashboard/frontend/src/app/shared/services/refresh-interval.service.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/shared/services/refresh-interval.service.ts
@@ -1,6 +1,6 @@
 import { Injectable } from '@angular/core';
 
-import { BehaviorSubject } from 'rxjs';
+import { BehaviorSubject, interval, Subscription } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
@@ -12,6 +12,7 @@ export class RefreshIntervalService {
   // Observable streams
   intervalData$ = this.intervalDataSource.asObservable();
 
+  private intervalSubscription: Subscription;
   private intervalTime: number;
   intervalID: number;
 
@@ -20,14 +21,16 @@ export class RefreshIntervalService {
     this.setRefreshInterval(initialInterval);
   }
 
-  setRefreshInterval(interval: number) {
-    this.intervalTime = interval;
-    sessionStorage.setItem('dashboard_interval', interval.toString());
+  setRefreshInterval(newInterval: number) {
+    this.intervalTime = newInterval;
+    sessionStorage.setItem('dashboard_interval', newInterval.toString());
 
-    window.clearInterval(this.intervalID);
-    this.intervalID = window.setInterval(() => {
-      this.intervalDataSource.next(this.intervalTime);
-    }, this.intervalTime);
+    if (this.intervalSubscription) {
+      this.intervalSubscription.unsubscribe();
+    }
+    this.intervalSubscription = interval(newInterval).subscribe(() =>
+      this.intervalDataSource.next(newInterval)
+    );
   }

```

Reply:
I will change it after @tspmelo  replied.

Reply:
Since we are already using rxjs, I'm ok with the change.
But `window.setInterval` is still a valid plain JS method :)

Reply:
@tspmelo I was discussing with @alfonsomthd that we _ideally_ should use a single approach (hopefully encapsulated into a single service) for handling all HTTP polling. Currently there's a mixture of `setInterval`, `Observable.timer`, recursive callbacks, etc. I was thinking on some registry service for pollers (that would make easier to implement observable caching and shared observables). For the feature toggles I cached a `shared` observable which allowed different parts of the DOM to seamlessly poll the same endpoint without duplicating queries (might look like an overkill there, but was needed for some corner case).